src/develop/pixelpipe_cache.c

```diff
@@ -179,54 +180,60 @@ uint64_t dt_dev_pixelpipe_cache_basichash_prior(
     pieces = g_list_next(pieces);
     modules = g_list_next(modules);
   }
-  return last>=0 ? dt_dev_pixelpipe_cache_basichash(imgid, pipe, last) : -1;
+  return (last >= 0) ? dt_dev_pixelpipe_cache_basichash(imgid, pipe, last) : -1;
 }
 
 void dt_dev_pixelpipe_cache_fullhash(
-        dt_imgid_t imgid,
+        const dt_imgid_t imgid,
         const dt_iop_roi_t *roi,
         struct dt_dev_pixelpipe_t *pipe,
-        int module,
+        const int position,
         uint64_t *basichash,
         uint64_t *fullhash)
 {
-  uint64_t hash = *basichash = dt_dev_pixelpipe_cache_basichash(imgid, pipe, module);
-  // also add scale, x and y:
+  uint64_t hash = *basichash = dt_dev_pixelpipe_cache_basichash(imgid, pipe, position);
+  // also include roi data
   const char *str = (const char *)roi;
   for(size_t i = 0; i < sizeof(dt_iop_roi_t); i++)
     hash = ((hash << 5) + hash) ^ str[i];
   *fullhash = hash;
 }
 
 uint64_t dt_dev_pixelpipe_cache_hash(
-           dt_imgid_t imgid,
+           const dt_imgid_t imgid,
            const dt_iop_roi_t *roi,
            dt_dev_pixelpipe_t *pipe,
-           int module)
+           const int position)
 {
   uint64_t basichash, hash;
-  dt_dev_pixelpipe_cache_fullhash(imgid, roi, pipe, module, &basichash, &hash);
+  dt_dev_pixelpipe_cache_fullhash(imgid, roi, pipe, position, &basichash, &hash);
   return hash;
 }
 
 gboolean dt_dev_pixelpipe_cache_available(
-           dt_dev_pixelpipe_cache_t *cache,
+           dt_dev_pixelpipe_t *pipe,
            const uint64_t hash,
            const size_t size)
 {
+  if(pipe->mask_display)
+    return FALSE;
+
+  dt_dev_pixelpipe_cache_t *cache = &(pipe->cache);
   // search for hash in cache and make the sizes are identical
   for(int k = 0; k < cache->entries; k++)
     if((cache->hash[k] == hash) && (cache->size[k] == size))
       return TRUE;
   return FALSE;
 }
 
+// While looking for the oldest cacheline we always ignore the first two lines as they are used
+// for swapping buffers while in entries==2 or masking mode
```

Comment:
Ok, but this 2 is used inside this routine and others later. Can we have a #define with a meaningful name (and set to 2) and use this consistently in the part where needed instead of literal 2 in all code below? This will help maintaining the code I think.

Reply:
Sure and you are right :-) Done with DT_PIPECACHE_MIN